src/main/java/jenkins/plugins/slack/SlackMessageRequest.java

```diff
@@ -0,0 +1,226 @@
+package jenkins.plugins.slack;
+
+import java.util.Objects;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang3.StringUtils;
+
+public class SlackMessageRequest implements SlackRequest {
```

Comment:
Much of the content of `StandardSlackService.publish(SlackRequest)` has been rehomed here. I tried to do that as faithfully as possible.